examples/pipeline_wavernn/losses.py

```diff
@@ -0,0 +1,101 @@
+import torch
+import torch.nn as nn
+from torch.nn import functional as F
+
+
+class MoLLoss(torch.nn.Module):
+    r""" Discretized mixture of logistic distributions loss
+
+    Adapted from wavenet vocoder
+    (https://github.com/r9y9/wavenet_vocoder/blob/master/wavenet_vocoder/mixture.py)
+    Explanation of loss (https://github.com/Rayhane-mamah/Tacotron-2/issues/155)
+
+    Args:
+        y_hat (Tensor): Predicted output (n_batch x n_time x n_channel)
+        y (Tensor): Target (n_batch x n_time x 1)
+        num_classes (int): Number of classes
+        log_scale_min (float): Log scale minimum value
+        reduce (bool): If True, the losses are averaged or summed for each minibatch
+
+    Returns
+        Tensor: loss
+    """
+
+    def __init__(self, num_classes=65536, log_scale_min=None, reduce=True):
+        super(MoLLoss, self).__init__()
+        self.num_classes = num_classes
+        self.log_scale_min = log_scale_min
+        self.reduce = reduce
+
+    def forward(self, y_hat, y):
+
+        if self.log_scale_min is None:
+            self.log_scale_min = torch.log(torch.as_tensor(1e-14)).item()
```

Comment:
nit: this doesn't really need torch. Would `math.log(1e-14)` do the same?

Reply:
yes, I tested in notebook and replaced it.